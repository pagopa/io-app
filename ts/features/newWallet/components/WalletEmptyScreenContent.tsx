import {
  Body,
  ButtonSolid,
  IOVisualCostants,
  Pictogram,
  VSpacer
} from "@pagopa/io-app-design-system";
import React from "react";
import { StyleSheet, View } from "react-native";
import I18n from "../../../i18n";
import { useIONavigation } from "../../../navigation/params/AppParamsList";
import { WalletRoutes } from "../navigation/routes";

const WalletEmptyScreenContent = () => {
  const navigation = useIONavigation();

  const handleAddToWalletButtonPress = () => {
    navigation.navigate(WalletRoutes.WALLET_NAVIGATOR, {
      screen: WalletRoutes.WALLET_CARD_ONBOARDING
    });
  };

  return (
    <View style={styles.container} testID="walletEmptyScreenContentTestID">
      <Pictogram name="cardAdd" />
      <VSpacer size={16} />
      <Body color="grey-650" weight="Regular" style={styles.text}>
        {I18n.t("features.wallet.home.emptyMessage")}
      </Body>
      <VSpacer size={24} />
      <ButtonSolid
        label={I18n.t("features.wallet.home.cta")}
        accessibilityLabel={I18n.t("features.wallet.home.cta")}
        onPress={handleAddToWalletButtonPress}
        icon="addSmall"
        iconPosition="end"
        fullWidth={true}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: IOVisualCostants.appMarginDefault,
    justifyContent: "center",
    alignItems: "center"
  },
  text: { textAlign: "center" }
});

export { WalletEmptyScreenContent };
