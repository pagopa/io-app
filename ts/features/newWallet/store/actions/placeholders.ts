import { ActionType, createStandardAction } from "typesafe-actions";

export const walletToggleLoadingState = createStandardAction(
  "WALLET_TOGGLE_LOADING_STATE"
)<boolean>();

export type WalletPlaceholdersActions = ActionType<
  typeof walletToggleLoadingState
>;
