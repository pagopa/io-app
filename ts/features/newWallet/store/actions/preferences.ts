import { ActionType, createStandardAction } from "typesafe-actions";

export const walletSetPaymentsRedirectBannerVisible = createStandardAction(
  "WALLET_SET_PAYMENTS_REDIRECT_BANNER_VISIBLE"
)<boolean>();

export type WalletPreferencesActions = ActionType<
  typeof walletSetPaymentsRedirectBannerVisible
>;
