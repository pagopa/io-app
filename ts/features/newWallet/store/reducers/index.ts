import { combineReducers } from "redux";
import { PersistPartial } from "redux-persist";
import preferencesPersistor, { WalletPreferencesState } from "./preferences";

export type WalletState = {
  preferences: WalletPreferencesState & PersistPartial;
};

const walletReducer = combineReducers({
  preferences: preferencesPersistor
});

export default walletReducer;
