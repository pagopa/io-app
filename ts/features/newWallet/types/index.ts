import { IOIcons } from "@pagopa/io-app-design-system";
import {
  IdPayWalletCard,
  IdPayWalletCardProps
} from "../../idpay/wallet/components/IdPayWalletCard";
import {
  PaymentWalletCard,
  PaymentWalletCardProps
} from "../../payments/common/components/PaymentWalletCard";
import { WalletCardBaseComponent } from "../components/WalletCardBaseComponent";
import {
  CgnWalletCard,
  CgnWalletCardProps
} from "../../bonus/cgn/components/CgnWalletCard";

// Used to group the cards in the wallet.
export type WalletCardCategory = "itw" | "cgn" | "bonus" | "payment";

// Basic type definition for a wallet card, describes the properties that
// each card MUST have in order to be placed inside the wallet.
type WalletCardBase = {
  key: string;
  category: WalletCardCategory;
};

// Specific type for ID Pay bonus cards
export type WalletCardBonus = {
  type: "idPay";
} & IdPayWalletCardProps;

// Specific type for CGN bonus cards
export type WalletCardCgn = {
  type: "cgn";
} & CgnWalletCardProps;

// Specific type for payment cards
export type WalletCardPayment = {
  type: "payment";
} & PaymentWalletCardProps;

// Base WalletCard type, which includes all card types
export type WalletCard = WalletCardBase &
  (WalletCardBonus | WalletCardCgn | WalletCardPayment);

// Used to map the card to the specific component that will render the card.
export type WalletCardType = WalletCard["type"];

/**
 * Wallet card component mapper which translates a WalletCardType to a
 * component to be rendered inside the wallet.
 * Component MUST be a WalletCardBaseComponent, which can be created
 * using {@see withWalletCardBaseComponent} HOC
 */
export const walletCardComponentMapper: Record<
  WalletCardType,
  WalletCardBaseComponent<any> | undefined
> = {
  cgn: CgnWalletCard,
  idPay: IdPayWalletCard,
  payment: PaymentWalletCard
};

/**
 * Icons used for each wallet card category
 */
export const walletCardCategoryIcons: Record<WalletCardCategory, IOIcons> = {
  bonus: "bonus",
  payment: "creditCard",
  itw: "fiscalCodeIndividual",
  cgn: "bonus"
};
