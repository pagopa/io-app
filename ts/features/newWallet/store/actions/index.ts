import { WalletPreferencesActions } from "./preferences";

export type WalletActions = WalletPreferencesActions;
