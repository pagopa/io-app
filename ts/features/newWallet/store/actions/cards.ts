import { ActionType, createStandardAction } from "typesafe-actions";
import { WalletCard } from "../../types";

export const walletUpsertCard =
  createStandardAction("WALLET_UPSERT_CARD")<WalletCard>();

export const walletAddCards =
  createStandardAction("WALLET_ADD_CARDS")<ReadonlyArray<WalletCard>>();

export const walletRemoveCards = createStandardAction("WALLET_REMOVE_CARDS")<
  ReadonlyArray<WalletCard["key"]>
>();

export type WalletCardsActions =
  | ActionType<typeof walletUpsertCard>
  | ActionType<typeof walletAddCards>
  | ActionType<typeof walletRemoveCards>;
