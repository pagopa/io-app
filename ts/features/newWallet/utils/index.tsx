import * as React from "react";
import { WalletCard, walletCardComponentMapper } from "../types";

export const renderWalletCardFn = (
  card: WalletCard,
  stacked: boolean = false
) => {
  const Component = walletCardComponentMapper[card.type];
  return Component ? (
    <Component
      testID={`walletCardTestID_${card.key}`}
      cardProps={card}
      isStacked={stacked}
    />
  ) : null;
};
