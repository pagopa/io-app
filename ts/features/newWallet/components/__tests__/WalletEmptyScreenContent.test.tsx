import React from "react";
import configureMockStore from "redux-mock-store";
import ROUTES from "../../../../navigation/routes";
import { applicationChangeState } from "../../../../store/actions/application";
import { appReducer } from "../../../../store/reducers";
import { GlobalState } from "../../../../store/reducers/types";
import { renderScreenWithNavigationStoreContext } from "../../../../utils/testWrapper";
import { WalletEmptyScreenContent } from "../WalletEmptyScreenContent";

describe("WalletEmptyScreenContent", () => {
  it("should match the snapshot", () => {
    const { component } = renderComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const renderComponent = () => {
  const globalState = appReducer(undefined, applicationChangeState("active"));

  const mockStore = configureMockStore<GlobalState>();
  const store: ReturnType<typeof mockStore> = mockStore(globalState);

  return {
    component: renderScreenWithNavigationStoreContext<GlobalState>(
      () => <WalletEmptyScreenContent />,
      ROUTES.WALLET_HOME,
      {},
      store
    ),
    store
  };
};
