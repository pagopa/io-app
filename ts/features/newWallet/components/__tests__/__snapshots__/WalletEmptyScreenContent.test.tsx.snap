// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletEmptyScreenContent should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  WALLET_HOME
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSModalScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                            "paddingHorizontal": 24,
                          }
                        }
                        testID="walletEmptyScreenContentTestID"
                      >
                        <RNSVGSvgView
                          align="xMidYMid"
                          bbHeight={120}
                          bbWidth={120}
                          color="#00C5CA"
                          focusable={false}
                          height={120}
                          meetOrSlice={0}
                          minX={0}
                          minY={0}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              Object {
                                "flex": 0,
                                "height": 120,
                                "width": 120,
                              },
                            ]
                          }
                          tintColor="#00C5CA"
                          vbHeight={240}
                          vbWidth={240}
                          width={120}
                        >
                          <RNSVGGroup
                            fill={
                              Object {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGPath
                              clipRule={0}
                              d="M30 120C45.464 120 58 107.464 58 92C58 86.9 56.6365 82.1184 54.2541 78H128.751L129.412 81.2325L132.448 85.7159L120.935 92.9714L109.803 101.358L104.322 107.432L104.171 112.246L108.65 115.669L118.52 112.847L127.859 106.712C127.859 106.712 133.541 101.728 133.791 101.978C134.052 102.238 134.232 107.713 134.232 107.713C134.232 107.713 135.204 112.376 135.575 112.336C135.945 112.306 141.146 114.518 141.146 114.518L143.07 122.114L149.893 128.609L159.022 129.88L163.34 127.618L165.675 122.634L164.843 116.119L183 115.188V174.964C183 179.407 179.393 183 174.954 183H17.0462C12.5972 183 9 179.397 9 174.964V110.521C9.38703 110.959 9.78759 111.386 10.201 111.799C15.268 116.866 22.268 120 30 120Z"
                              fill={
                                Object {
                                  "payload": 4289392367,
                                  "type": 0,
                                }
                              }
                              fillRule={0}
                              propList={
                                Array [
                                  "fill",
                                  "fillRule",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M32 78H28V89H17V93H28V105H32V93H44V89H32V78Z"
                              fill={
                                Object {
                                  "payload": 4278240714,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                            <RNSVGPath
                              clipRule={0}
                              d="M15 115.647C14.3124 115.21 13.6451 114.744 13 114.25V171.576C13 175.672 16.4318 179 20.6762 179H171.324C175.559 179 179 175.681 179 171.576V116.355L161.678 117.215L164.587 123.641L161.678 129.927H156.124L147.416 128.753L142.719 122.172L140.282 117.874C140.11 117.143 139.278 116.064 139.242 116.039L139.07 115.736C139.07 115.736 135.617 113.326 135.263 113.354C134.91 113.391 133.772 109.434 133.772 109.434C133.772 109.434 133.218 107.535 132.733 106.012L132.731 106.004C132.426 105.049 132.149 104.244 132.054 104.151C131.971 104.072 131.27 104.569 130.388 105.254L130.371 105.267C128.701 106.565 126.394 108.524 126.394 108.524L117.485 114.192L108.069 116.799L117.485 114.621L127.584 110.138L127.69 110.048L127.748 109.999L127.917 109.856C128.064 109.732 128.275 109.555 128.528 109.343C129.035 108.919 129.712 108.357 130.393 107.805C130.611 107.628 130.828 107.454 131.038 107.287C131.047 107.318 131.246 107.275 131.445 107.232C131.645 107.189 131.845 107.145 131.854 107.177C131.96 107.52 131.843 107.948 131.735 108.344C131.643 108.681 131.557 108.995 131.622 109.213C131.692 109.449 131.75 109.645 131.79 109.782L131.848 109.981L131.85 109.987L131.851 109.988L131.852 109.991L131.852 109.994L131.854 110.001L131.865 110.037L131.903 110.168C131.936 110.281 131.984 110.441 132.042 110.632C132.157 111.012 132.316 111.525 132.488 112.041C132.577 112.306 132.799 112.435 133.018 112.562C133.215 112.677 133.41 112.789 133.5 113C133.59 113.21 133.71 113.472 133.856 113.707C133.924 113.816 134.054 114.015 134.253 114.208C134.335 114.287 134.662 114.601 135.185 114.742C135.201 114.751 135.26 114.744 135.332 114.736C135.55 114.712 135.887 114.674 135.5 115C135.661 115.096 135.769 115.203 135.887 115.321C136.012 115.446 136.151 115.584 136.38 115.736C136.616 115.891 137.323 115.904 137.981 115.916C138.555 115.926 139.091 115.936 139.242 116.039L140.282 117.874C140.389 118.326 140.243 118.645 139.5 118.5L141.03 123.249L147.416 129.5L155.99 131.927L162.5 131L166.786 123.651L164.587 118L177 117.5V171.576C177 174.512 174.519 177 171.324 177H20.6762C17.4745 177 15 174.506 15 171.576V115.647ZM56.8405 84H129.046L130.5 86.5L118.694 94.1538L107.835 102.075L101.964 108.376L103.804 113.376L103.796 113.637L103.939 113.743L103.804 113.376L103.939 109.19L109.168 103.578L119.789 95.8307L131.784 88.8586L130.773 82H56.1615C56.4121 82.6552 56.6388 83.3222 56.8405 84Z"
                              fill={
                                Object {
                                  "payload": 4278240714,
                                  "type": 0,
                                }
                              }
                              fillRule={0}
                              propList={
                                Array [
                                  "fill",
                                  "fillRule",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M108.86 117.92C106.37 117.92 104.05 116.68 102.67 114.5C100.97 111.84 101.13 108.49 103.07 105.98C114.57 91.05 145.16 78.31 146.46 77.78L146.73 77.69C188.93 66.99 208.32 56.11 208.51 56L210.49 59.47C209.7 59.92 190.57 70.69 147.86 81.53C145.68 82.45 116.73 94.81 106.24 108.42C105.15 109.83 105.43 111.38 106.05 112.35C106.67 113.32 107.94 114.21 109.66 113.83C118.64 111.83 131.66 105.87 142.68 88.88L146.04 91.05C134.29 109.18 120.25 115.56 110.54 117.73C109.98 117.85 109.43 117.91 108.88 117.91L108.86 117.92Z"
                              fill={
                                Object {
                                  "payload": 4278927075,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M139.07 117.01C136.19 116.44 134.08 115.02 132.82 112.79C130.25 108.27 132.38 102.11 132.48 101.85L136.25 103.18C135.79 104.48 134.92 108.39 136.31 110.82C136.99 112 138.15 112.74 139.86 113.08L139.08 117L139.07 117.01Z"
                              fill={
                                Object {
                                  "payload": 4278927075,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M156.99 132.98C152.74 132.98 147.75 130.95 143.1 126.09C139.43 122.26 137.94 118.26 138.66 114.18C140.13 105.9 150.45 100.76 150.89 100.55L152.65 104.14C152.56 104.18 143.71 108.61 142.6 114.89C142.12 117.63 143.26 120.47 145.99 123.32C152.01 129.6 158.56 130.11 161.67 127.39C164.22 125.16 163.87 121.19 160.75 117.29L163.88 114.8C169.07 121.3 167.83 127.32 164.3 130.41C162.43 132.05 159.88 132.99 156.99 132.99V132.98Z"
                              fill={
                                Object {
                                  "payload": 4278927075,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M165.2 118.63C158.65 118.63 154.44 118.19 154.15 118.15L154.58 114.17C155.04 114.22 201.36 118.92 227.78 95.63L230.42 98.63C222.39 105.71 207.41 114.77 181.67 117.7C175.4 118.41 169.73 118.63 165.19 118.63H165.2Z"
                              fill={
                                Object {
                                  "payload": 4278927075,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                            <RNSVGPath
                              d="M132.12 87.65C131.47 87.19 125.8 83.03 126.1 78.11C126.25 75.61 127.82 73.54 130.77 71.95C140.2 66.87 170.74 70.1 172.04 70.24L171.61 74.22C163.13 73.3 139.59 71.74 132.67 75.47C131.02 76.36 130.16 77.33 130.09 78.35C129.96 80.5 132.96 83.34 134.43 84.38L132.12 87.65Z"
                              fill={
                                Object {
                                  "payload": 4278927075,
                                  "type": 0,
                                }
                              }
                              propList={
                                Array [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGSvgView>
                        <View
                          style={
                            Object {
                              "height": 16,
                            }
                          }
                        />
                        <Text
                          allowFontScaling={false}
                          color="grey-650"
                          defaultColor="bluegrey"
                          defaultWeight="Regular"
                          font="TitilliumWeb"
                          fontStyle={
                            Object {
                              "fontSize": 16,
                              "lineHeight": 24,
                            }
                          }
                          style={
                            Array [
                              Object {
                                "textAlign": "center",
                              },
                              Object {
                                "fontSize": 16,
                                "lineHeight": 24,
                              },
                              Object {
                                "color": "#636B82",
                                "fontFamily": "Titillium Web",
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                            ]
                          }
                          weight="Regular"
                        >
                          Custodisci qui i tuoi metodi di pagamento, Carta Giovani Nazionale, bonus e sconti.
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 24,
                            }
                          }
                        />
                        <View
                          accessibilityLabel="Aggiungi al Portafoglio"
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "busy": false,
                              "disabled": false,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={Object {}}
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "borderRadius": 4,
                                  "elevation": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "paddingHorizontal": 16,
                                  "textAlignVertical": "center",
                                },
                                Object {
                                  "overflow": "hidden",
                                },
                                false,
                                Object {
                                  "height": 40,
                                },
                                Object {
                                  "backgroundColor": "#0073E6",
                                },
                                Object {
                                  "backgroundColor": undefined,
                                  "transform": Array [
                                    Object {
                                      "scale": undefined,
                                    },
                                  ],
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "flexDirection": "row-reverse",
                                  },
                                ]
                              }
                            >
                              <RNSVGSvgView
                                accessibilityElementsHidden={true}
                                accessibilityLabel=""
                                accessible={false}
                                align="xMidYMid"
                                bbHeight={20}
                                bbWidth={20}
                                focusable={false}
                                height={20}
                                importantForAccessibility="no-hide-descendants"
                                meetOrSlice={0}
                                minX={0}
                                minY={0}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "borderWidth": 0,
                                    },
                                    Object {
                                      "color": "#FFFFFF",
                                    },
                                    Object {
                                      "flex": 0,
                                      "height": 20,
                                      "width": 20,
                                    },
                                  ]
                                }
                                tintColor="#FFFFFF"
                                vbHeight={24}
                                vbWidth={24}
                                width={20}
                              >
                                <RNSVGGroup
                                  fill={
                                    Object {
                                      "payload": 4278190080,
                                      "type": 0,
                                    }
                                  }
                                >
                                  <RNSVGPath
                                    clipRule={0}
                                    d="M12.9999 6C12.9999 5.44772 12.5522 5 11.9999 5C11.4477 5 10.9999 5.44772 10.9999 6V11L6.00098 11C5.44869 11 5.00098 11.4477 5.00098 12C5.00098 12.5523 5.44869 13 6.00098 13L10.9999 13V18.0008C10.9999 18.553 11.4477 19.0008 11.9999 19.0008C12.5522 19.0008 12.9999 18.553 12.9999 18.0008V13L17.9998 13C18.5521 13 18.9998 12.5523 18.9998 12C18.9998 11.4477 18.5521 11 17.9998 11L12.9999 11V6Z"
                                    fill={
                                      Object {
                                        "type": 2,
                                      }
                                    }
                                    fillRule={0}
                                    propList={
                                      Array [
                                        "fill",
                                        "fillRule",
                                      ]
                                    }
                                  />
                                </RNSVGGroup>
                              </RNSVGSvgView>
                              <View
                                style={
                                  Object {
                                    "width": 8,
                                  }
                                }
                              />
                              <Text
                                accessibilityElementsHidden={true}
                                accessible={false}
                                allowFontScaling={false}
                                color="white"
                                defaultColor="white"
                                defaultWeight="Bold"
                                ellipsizeMode="tail"
                                font="TitilliumWeb"
                                fontStyle={
                                  Object {
                                    "fontSize": 16,
                                  }
                                }
                                importantForAccessibility="no-hide-descendants"
                                maxFontSizeMultiplier={1.3}
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "alignSelf": "center",
                                    },
                                    Object {
                                      "fontSize": 16,
                                    },
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Titillium Web",
                                      "fontStyle": "normal",
                                      "fontWeight": "700",
                                    },
                                  ]
                                }
                                weight="Bold"
                              >
                                Aggiungi al Portafoglio
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSModalScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
